packages/hash/api/src/model/knowledge/link.model.ts

```diff
@@ -0,0 +1,160 @@
+
+import { EntityModel, LinkModel, LinkTypeModel } from "../index";
+
+export type LinkModelConstructorParams = {
+  accountId: string;
+  sourceEntity: EntityModel;
+  linkTypeModel: LinkTypeModel;
+  targetEntity: EntityModel;
+};
+
+export type LinkModelCreateParams = {
+  accountId: string;
+  sourceEntity: EntityModel;
+  linkTypeModel: LinkTypeModel;
+  targetEntity: EntityModel;
+};
+
+/**
+ * @class {@link LinkModel}
+ */
+export default class {
+  accountId: string;
+
+  sourceEntity: EntityModel;
+  linkTypeModel: LinkTypeModel;
+  targetEntity: EntityModel;
+
+  constructor({
+    accountId,
+    sourceEntity,
+    linkTypeModel,
+    targetEntity,
+  }: LinkModelConstructorParams) {
+    this.accountId = accountId;
+
+    this.sourceEntity = sourceEntity;
+    this.linkTypeModel = linkTypeModel;
+    this.targetEntity = targetEntity;
+  }
+
+  /**
+   * Create a link between a source and a target entity using a specific link
+   * type.
+   *
+   * @param params.accountId the accountId of the account creating the link
+   * @param params.sourceEntity the source entity of the link
+   * @param params.linkTypeModel the Link Type of the link
+   * @param params.targetEntity the target entity of the link
+   */
+  static async create(
+    graphApi: GraphApi,
+    {
+      accountId,
+      sourceEntity,
+      linkTypeModel,
+      targetEntity,
+    }: LinkModelCreateParams,
+  ): Promise<LinkModel> {
+    const {
+      data: { sourceEntityId, linkTypeUri, targetEntityId },
+    } = await graphApi.createLink(sourceEntity.entityId, {
+      accountId,
+      linkTypeUri: linkTypeModel.schema.$id,
+      targetEntityId: targetEntity.entityId,
+    });
+
+    const fetchedParams = {
+      sourceEntity: await EntityModel.getLatest(graphApi, {
+        accountId,
+        entityId: sourceEntityId,
+      }),
+      linkTypeModel: await LinkTypeModel.get(graphApi, {
+        versionedUri: linkTypeUri,
+      }),
+      targetEntity: await EntityModel.getLatest(graphApi, {
+        accountId,
+        entityId: targetEntityId,
+      }),
+    };
+
+    return new LinkModel({
+      accountId,
+      ...fetchedParams,
+    });
+  }
+
+  /**
+   * Get all outgoing links of a source entity.
+   *
+   * @param params.accountId the accountId of the account creating the link
+   * @param params.sourceEntity the source entity of the link
+   */
+  static async getAllOutgoing(
+    graphApi: GraphApi,
+    { sourceEntity }: { sourceEntity: EntityModel },
+  ): Promise<LinkModel[]> {
+    const { data: entityLinks } = await graphApi.getEntityLinks(
+      sourceEntity.entityId,
+    );
+    return Promise.all(
+      Object.entries(entityLinks.outgoing)
+        /**
+         * When target entity id is a list the link has many targets.
+         * This is not supported yet
+         * @todo support 1:N links
+         * */
+        .filter(([_, targetEntityId]) => typeof targetEntityId === "string")
```

Comment:
Why is this necessary

Reply:
The endpoint returns `{ [string]: string | string[] }`, so this is to only deal with the singular links

Reply:
Gotcha, I didn't realise it was filtering out arrays. Wasn't sure if we had other stuff in here like `undefined` or some other stuff which felt like it should have been handled somewhere else haha